holidays/countries/thailand.py

```diff
@@ -12,156 +12,633 @@
+    - [Makha Bucha]
+        https://www.onab.go.th/th/content/category/detail/id/73/iid/3403
+    - [Visakha Bucha]
+        https://www.onab.go.th/th/content/category/detail/id/73/iid/3401
+    - [Asarnha Bucha]
+        https://www.onab.go.th/th/content/category/detail/id/73/iid/3397
+    - [Buddhist Lent Day]
+        https://www.onab.go.th/th/content/category/detail/id/73/iid/3395
+    - [Royal Ploughing Ceremony]
+        https://en.wikipedia.org/wiki/Royal_Ploughing_Ceremony
+        https://www.lib.ru.ac.th/journal/may/may_phauchmongkol.html
+        https://www.myhora.com/ปฏิทิน/ปฏิทิน-พ.ศ.2540.aspx
+
+
+    Limitations:
+
+    - This is only 100% accurate for 1997-2023; any future dates are up to the
+      Royal Thai Government Gazette which updates on a year-by-year basis.
+
+    - Approx. date only goes as far back as 1941 (B.E. 2484) as the Thai
+      calendar for B.E. 2483 as we only have nine months from switching
+      New Year Date (April 1st to January 1st).
+
+    - Thai Lunar Calendar Holidays only work from 1941 (B.E. 2484) onwards
+      until 2057 (B.E. 2600) as we only have Thai year-type data for
+      cross-checking until then.
+
+    - Royal Ploughing Ceremony Day is date is announced on an annual basis
+      by the Court Astrologers, thus need an annual update to the library here
+
+    - This doesn't cover Thai regional public holidays yet, only stubs added
+
+
+    Country created by: `arkid15r <https://github.com/arkid15r>`__
+
+    Country maintained by: `PPsyrius <https://github.com/PPsyrius>`__
     """
 
     country = "TH"
 
-    def __init__(self, **kwargs):
-        self.cnls = _ChineseLuniSolar()
-        HolidayBase.__init__(self, **kwargs)
+    # วันหยุดพิเศษ (เพิ่มเติม) - see Bank of Thailand's DB for Cross-Check
+    thai_special_in_lieu_holidays = "Special In Lieu Holiday"
+    thai_election = "Thai Election Day"
+    thai_election_in_lieu = "Thai Election Day (in lieu)"
+    thai_bridge_public_holiday = "Bridge Public Holiday"
+
+    # Special Cases
+    rama_ix_golden_jubilee = "HM King Bhumibol Adulyadej's Golden Jubilee"
+    rama_ix_sixty_accession = (
+        "HM King Bhumibol Adulyadej's 60th Anniversary of Accession Event"
+    )
+    thai_military_emergency_lockdown = (
+        "Emergency Lockdown (Thai Military Coup d'état)"
+    )
+    thai_political_emergency_lockdown = (
+        "Emergency Lockdown (Thai Political Unrest)"
+    )
+    thai_flood_2011_emergency_lockdown = (
+        "Emergency Lockdown (2011 Thailand Floods)"
+    )
+    rama_ix_mourning = "Day of Mourning for HM King Bhumibol Adulyadej"
+    rama_ix_cremation = "HM King Bhumibol Adulyadej's Royal Cremation Ceremony"
+    rama_x_coronation_celebrations = (
+        "HM King Maha Vajiralongkorn's Coronation Celebrations"
+    )
+    songkran_festival_in_lieu_covid = "Songkran Festival (in lieu)"
+
+    special_holidays = {
+        # 1992-1994 (include In Lieus, Checked /w Bank of Thailand Data)
```

Comment:
It looks like '/w' should be replaced with 'w/'?

Reply:
I did gone with `with` itself instead for now